source/extensions/filters/network/redis_proxy/command_splitter_impl.h

```diff
@@ -75,13 +85,19 @@ class SplitRequestBase : public SplitRequest {
                                        const Common::Redis::RespValue& request);
   void updateStats(const bool success);
 
+  // To support delay faults, we allow faults to override the regular command latency
+  // recording behavior.
+  void delayLatencyMetric() { delay_command_latency_ = true; }
```

Comment:
why do we need these new interface on the base class instead of just in the delay fault?

Reply:
The request that is being delayed records its latency before calling the  onResponse() callback. That means unless the request is somehow aware that it should not calculate it and whatever is on the other side of the callback will, we will generate the wrong latency data.

In a  request the `updateStats()` call always precedes `callbacks.onResponse()`:

```
void SplitRequestBase::updateStats(const bool success) {
  if (success) {
    command_stats_.success_.inc();
  } else {
    command_stats_.error_.inc();
  }
  if (!delay_command_latency_) {
    command_latency_->complete();
  }
}
```

Does that make sense?



Does that make sense?

Request.